yt_dlp/extractor/cbc.py

```diff
@@ -209,289 +198,232 @@ def _real_extract(self, url):
-                })
-
-            timestamp = None
-            release_date = find_xpath_attr(
-                item, _add_ns('media:credit'), 'role', 'releaseDate')
-            if release_date is not None:
-                timestamp = parse_iso8601(release_date.text)
-
-            entries.append({
-                '_type': 'url_transparent',
-                'url': content_url,
-                'id': guid,
-                'title': title,
-                'description': xpath_text(item, 'description'),
-                'timestamp': timestamp,
-                'duration': int_or_none(content.get('duration')),
-                'age_limit': parse_age_limit(xpath_text(item, _add_ns('media:rating'))),
-                'episode': xpath_text(item, _add_ns('clearleap:episode')),
-                'episode_number': int_or_none(xpath_text(item, _add_ns('clearleap:episodeInSeason'))),
-                'series': xpath_text(item, _add_ns('clearleap:series')),
-                'season_number': int_or_none(xpath_text(item, _add_ns('clearleap:season'))),
-                'thumbnails': thumbnails,
-                'ie_key': 'CBCWatchVideo',
-            })
-
-        return self.playlist_result(
-            entries, xpath_text(channel, 'guid'),
-            xpath_text(channel, 'title'),
-            xpath_text(channel, 'description'))
-
-
-class CBCWatchVideoIE(CBCWatchBaseIE):
-    IE_NAME = 'cbc.ca:watch:video'
-    _VALID_URL = r'https?://api-cbc\.cloud\.clearleap\.com/cloffice/client/web/play/?\?.*?\bcontentId=(?P<id>[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12})'
-    _TEST = {
+class CBCGemIE(InfoExtractor):
+    IE_NAME = 'gem.cbc.ca'
+    _VALID_URL = r'https?://gem\.cbc\.ca/media/(?P<id>[0-9a-z-]+/s[0-9]+[a-z][0-9]+)'
+    _TESTS = [{
         # geo-restricted to Canada, bypassable
-        'url': 'https://api-cbc.cloud.clearleap.com/cloffice/client/web/play/?contentId=3c84472a-1eea-4dee-9267-2655d5055dcf&categoryId=ebc258f5-ee40-4cca-b66b-ba6bd55b7235',
-        'only_matching': True,
-    }
+        # This is a normal, public, TV show video
+        'url': 'https://gem.cbc.ca/media/schitts-creek/s06e01',
+        'md5': '93dbb31c74a8e45b378cf13bd3f6f11e',
+        'info_dict': {
+            'id': 'schitts-creek/s06e01',
+            'ext': 'mp4',
+            'title': 'Smoke Signals',
+            'description': 'md5:929868d20021c924020641769eb3e7f1',
+            'thumbnail': 'https://images.radio-canada.ca/v1/synps-cbc/episode/perso/cbc_schitts_creek_season_06e01_thumbnail_v01.jpg?im=Resize=(Size)',
+            'duration': 1314,
+            'categories': ['comedy'],
+            'series': 'Schitt\'s Creek',
+            'season': 'Season 6',
+            'season_number': 6,
+            'episode': 'Smoke Signals',
+            'episode_number': 1,
+            'episode_id': 'schitts-creek/s06e01',
+        },
+        'params': {'format': 'bv'},
+        'skip': 'Geo-restricted to Canada',
```

Comment:
The original comment above says the geo-restriction is bypassable?

Reply:
That kind of comment was already in this code when I made this PR. I assume it refers to the fact that's it's possible to bypass the geo-restriction with proxies. It's not possible to bypass it in this code.

Reply:
Yes, testing the Schitt URLs after posting the above comment, as part of https://github.com/ytdl-org/youtube-dl/pull/30046 (and thanks again), they get two tries, and the geo-bypass on the second try fails.

In that case I think it's correct to delete the comment (here and below, where applicable) as the comment must have meant that geo-bypass was possible.

Reply:
Geo-bypass is possible, by using proxies. I'm not sure what you mean.

Reply:
"Geo-bypass" in youtube-dl has a very specific meaning of using the X-Forwarded-For header trick, which only works on a few sites. Using a proxy doesn't really count as bypassing here, even if in general calling it that makes sense, because you're not *actually* in the required country.

Reply:
Oh I see, my bad. Then yes, the comment should be removed I guess. Feel free to do it in your PR @dirkf

Reply:
Apparently the Schitt playlist in the test for `CBCGemPlaylistIE` is not restricted: my version of the test runs fine in the UK without a `'skip'`, and with no geo-bypassing added:
```
# python test/test_download.py TestDownload.test_CBCGemPlaylist
[cbc.ca:gem:playlist] schitts-creek/s06: Downloading JSON metadata
[download] Downloading playlist: Season 6
[cbc.ca:gem:playlist] playlist Season 6: Collected 16 video ids (downloading 16 of them)
[download] Downloading video 1 of 16
...
[download] Downloading video 16 of 16
[download] Finished downloading playlist: Season 6
.
----------------------------------------------------------------------
Ran 1 test in 54.818s

OK
# 
```

Reply:
Ah that makes sense actually. Probably everything else will fail though.

Reply:
I have made the required changes in another PR, #1162. Feel free to make further comments there.